manual/src/main/asciidoc/jsonSchema/introduction.adoc

```diff
@@ -0,0 +1,280 @@
+    "$schema": "https://json-schema.org/draft/2019-09/schema",
+    "type": "object",
+    "properties": {
+        "fromAllOf": {
+            "type": "string",
+            "$comment": "Example of allOf."
+        }
+    }
+}
+----
+
+With the following data, the validation will fail because the property *myNewProperty* is not defined neither the *properties* part nor the *allOf* part.
+
+[source]
+----
+{
+    "myProperty": "My property",
+    "fromAllOf" : 10,
+    "myNewProperty": "another one" //Not valid
+}
+----
+
+==== How are JSON Schema used in Unomi
+
+JSON Schema is used in UNOMI to validate the data coming from the two public endpoints */contextRequest* and */eventCollector*.
+Both endpoints have a custom deserializer which will begin by validating the payload of the request, then will filter invalid events present in this payload.
+If an event is not valid it will not be processed by the system.
+The internal events are not validated by JSON schema as they are not sent through the public endpoints.
+
+In UNOMI, each event type must have an associated JSON schema.
+To validate an event, UNOMI will search for a schema in which the target of the schema is *events*, and with the name of the schema matching the event type.
+
+A custom keyword named *self* has to be present in the JSON schemas to store the information related to each schema.
+The following example is the *self* part of the view event JSON schema.
+Having the target set to *events* and the name set to *view*, this schema will be used to validate the events of type *view*.
+
+[source]
+----
+…
+"self":{
+    "vendor":"org.apache.unomi",
+    "target" : "events",
+    "name": "view",
+    "format":"jsonschema",
+    "version":"1-0-0"
+},
+…
+----
+
+Link to the schema on github: https://github.com/apache/unomi/blob/master/extensions/json-schema/services/src/main/resources/META-INF/cxs/schemas/events/view/view.json[view.json].
+
+A set of predefined schema are present in UNOMI, these schemas can be found under the folder : https://github.com/apache/unomi/tree/master/extensions/json-schema/services/src/main/resources/META-INF/cxs/schemas[extensions/json-schema/services/src/main/resources/META-INF/cxs/schemas].
+
+These schemas will be loaded in memory at startup.
+Each schema where the *target* value is set to *events*, will be used to validate events.
+The others are simply used as part of JSON schema or can be used in additional JSON schemas.
+
+It’s possible to add JSON schema to validate your own event by using the API, the explanations to manage JSON schema through the API are
+in the <<Create / update a JSON schema to validate an event, Create / update a JSON schema to validate an event>> section.
+
+Contrary to the predefined schemas, the schemas added through the API will be persisted in elasticsearch in the jsonSchema index.
+Schemas persisted in Elasticsearch do not need a restart of the platform to reflect changes.
```

Comment:
```suggestion
Schemas persisted in Elasticsearch do not need a require of the platform to reflect changes.
```

Reply:
I think you wanted to say ... "do not require a restart..."